packages/terra-button/src/Button.scss

```diff
@@ -117,4 +117,12 @@
     padding-bottom: var(--terra-button-padding-bottom-compact, 0);
     padding-top: var(--terra-button-padding-top-compact, 0);
   }
+
+  .text-with-icon {
+    padding-left: 0.714em;
+  }
+
+  .text-with-icon-reverse {
+    padding-right: 0.714em;
+  }
```

Comment:
I think these should be margins to push space between content

Reply:
IMO we should have a component to manage this. Badges do basically the exact same thing.

Reply:
I'm working the OCS clinical/consumer teams to create a component for this. The consumer UX team started design work on this today.

Reply:
Changed to margins.

Looking forward to the component.